relayer/chains/evm/listener.go

```diff
@@ -81,120 +70,82 @@ func (p *Provider) Listener(ctx context.Context, lastSavedHeight uint64, blockIn
 		case <-subscribeStart.C:
 			subscribeStart.Stop()
 			go p.Subscribe(ctx, blockInfoChan, resetFunc)
-		case <-pollHeightTicker.C:
-			pollHeightTicker.Stop()
-			next = p.GetLastSavedBlockHeight()
-			latest = p.latestHeight()
-
-		case bn := <-bnch:
-			// process all notifications
-			for ; bn != nil; next++ {
-				if lbn != nil {
-					p.log.Debug("block-notification received", zap.Uint64("height", lbn.Height.Uint64()),
-						zap.Int64("gas-used", int64(lbn.Header.GasUsed)))
-
-					messages, err := p.FindMessages(ctx, lbn)
-					if err != nil {
-						return errors.Wrapf(err, "receiveLoop: callback: %v", err)
-					}
-					blockInfoChan <- &relayertypes.BlockInfo{
-						Height:   lbn.Height.Uint64(),
-						Messages: messages,
-					}
-				}
 
-				if lbn, bn = bn, nil; len(bnch) > 0 {
-					bn = <-bnch
-				}
-			}
-			// remove unprocessed notifications
-			for len(bnch) > 0 {
-				<-bnch
-			}
-
-		default:
-			if next >= latest {
-				continue
+			if isSubError {
+				startHeight = p.GetLastSavedBlockHeight()
 			}
 
-			qch := make(chan *bnq, cap(bnch))
-			for i := next; i < latest && len(qch) < cap(qch); i++ {
-				qch <- &bnq{i, nil, nil, 3} // fill bch with requests
+			var blockReqs []*blockReq
+			for i := startHeight; i < latestHeight; i += p.cfg.BlockBatchSize {
```

Comment:
Latest height in this expression is never updated as the time progresses. in this case fallback polling for missed heights from WebSockets  won't work.

Reply:
yep, we don't need to update latest height. It's triggered once when subs errors and exits.

Reply:
how do we know if something went wrong in this particular set of blocks when required manual intervention?

Reply:
we don't know unless reported by users. We have production support on plan.